### ISSUE TITLE
Internal repos are treated as private for Github, confusing customers

### ISSUE THREAD

**Issue by kopancek** [sourcegraph/sourcegraph#56031]
TLDR: This is a followup issue to multiple customer issues that were reported for GitHub repos internal repository visibility to users. The most recent one being here: https://sourcegraph.slack.com/archives/C05EMJM2SLR/p1692356218740489?thread_ts=1691526938.978639&cid=C05EMJM2SLR

> Internal repos that people don't have explicit access to aren't added to a user's accessible repo list
I'm testing a possible fix here where we check for internal repositories in the permissions sync, but it comes with another issue: There's no way to do a repo-centric permissions sync then, since the internal repos are basically "public" and you can't ask for all collaborators on the internal repository
What we could do is do the is:internal repo visibility trick:
Ask them to set up a second code host connection, copying the first one exactly, but with these differences:
Take out the authorization: {}
Take out the cloneAllRepositories: true
Add in: "repositoryQuery": ["org:krogertechnology is:internal"]

This issue is mostly about gathering information, feedback and coming up with a longer term solution.

Treating the internal repos as private originated in #42424
- #42424

The current problems with internal repos treated as private are at least that:
1. We need to sync permissions for these repos. Since all users of Sourcegraph usually have access to the repos, we unnecessarily increase the size of the permissions tables, making syncing and perms enforcing slower
2. Github API itself is not consistent in how internal repositories are treated.
3. Setting up another code host connection for internal repos seems like a hack and it's not documented in our docs.

There is a related PR #55404 that tries to fix the problem by adding a toggle for treating internal repos as public.
- #55404

Maybe we should make that the default and only enforce internal repo visibility if customers actually turn it on? 🤔 

/cc @sourcegraph/source

**Comment by mrnugget** [comment]
@pjlast says it's fixed
